--- private/worker.h
+++ public/worker.h
@@ -47,7 +47,7 @@
                 worker(config * conf_obj, torrent_list &torrents, user_list &users, std::vector<std::string> &_whitelist, mysql * db_obj, site_comm * sc);
                 void reload_config(config * conf);
                 std::string work(const std::string &input, std::string &ip, client_opts_t &client_opts);
-                std::string announce(torrent &tor, user_ptr &u, params_type &params, params_type &headers, std::string &ip, client_opts_t &client_opts);
+                std::string announce(const std::string &input, torrent &tor, user_ptr &u, params_type &params, params_type &headers, std::string &ip, client_opts_t &client_opts);
                 std::string scrape(const std::list<std::string> &infohashes, params_type &headers, client_opts_t &client_opts);
                 std::string update(params_type &params, client_opts_t &client_opts);
 
